packages/shared/lib/contexts/onboarding/actions/initialiseOnboarding.ts

```diff
@@ -0,0 +1,27 @@
+import { get } from 'svelte/store'
+
+import { initialiseOnboardingRouters, OnboardingRoute, onboardingRouter } from '@core/router'
+
+import { IOnboardingInitialisationOptions } from '../interfaces'
+import { updateOnboardingProfile } from '../stores'
+
+import { initialiseOnboardingProfile } from './initialiseOnboardingProfile'
+import { logout } from '@core/profile'
+
+export async function initialiseOnboarding(options: IOnboardingInitialisationOptions): Promise<void> {
+    const { isDeveloperProfile, networkProtocol, networkType, resetRouters } = options
+
+    await logout()
+
+    initialiseOnboardingProfile(isDeveloperProfile, networkProtocol, true)
+
+    if (networkType) {
+        updateOnboardingProfile({ networkType })
+    }
+
+    if (resetRouters) {
```

Comment:
When would we not reset the router? Can we not just do this in all cases as we want to be at the start of the correct onboarding flow no matter what.